Crud.php

```diff
@@ -0,0 +1,271 @@
+<?php
+    use Doctrine\ORM\Tools\Pagination\Paginator;
+    require_once('src/Number.php');
+
+    /**
+     * Class to handle main actions(CRUD) of web app
+     */
+    class Crud {
+        //Object for manage doctrine entities
+        private $entityManager;
+
+        const PER_PAGE = 5; // Const for paginations(items per page)
+        const ENTITY_NAME = "Number"; // NAme of entity this controller handle
+        const ENTITY_PATH = __DIR__."/src/".self::ENTITY_NAME; // PAth to entity file
+        const ENTITY_ALIAS = "src\Number"; // Alias for entity
+
+        /**
+         * Method for creating of class
+         * 
+         * Takes entyty manager instance to work with doctrine
+         * entity manager instance is generated inside bootstrap file
+         * 
+         * @param entityManager
+         */
+        public function __construct($entityManager) {
+            $this->entityManager = $entityManager;
+        }
+
+        /**
+         * Adds new number
+         * 
+         * @param string slug
+         * @param string title
+         * @param int number
+         * @param string text
+         * @param string transcription
+         * 
+         * @return bool
+         */
+        public function createNumber(string $slug, string $title, int $number, string $text, string $transcription):bool {
+            $numberEntity = new Number();
+            $numberEntity->setSlug($slug);
+            $numberEntity->setTitle($title);
+            $numberEntity->setNumber($number);
+            $numberEntity->setText($text);
+            $numberEntity->setTranscription($transcription);
+            $numberEntity->setDate(new DateTime());
+            $this->entityManager->persist($numberEntity);
+            $this->entityManager->flush();
+
+            return true;
+        }
+
+        /**
+         * Read number data
+         * 
+         * @param int id
+         * 
+         * @return object
+         */
+        public function readNumber(int $id):object {
+            $number = $this->entityManager->find('Number', $id);
+
+            return $number;
+        }
+
+        /**
+         * Search numbers
+         * 
+         * @param slug
+         * @param title
+         * @param text
+         * @param transcription
+         * @param date
+         */
+        public function findNumbers($slug = null, $title = null, $text = null, $transcription = null, $date = null) {}
+
+        /**
+         * Read all numbers
+         * 
+         * @return object
+         */
+        public function readAllNumbers():object {
+            $numberRepository = $this->entityManager->getRepository('Number');
```

Comment:
What is the difference between `$this->entityManager->getRepository('Number')` and `$this->entityManager->getRepository(Number::class)`

Reply:
No difference. Will use Number::class because it use internal data from class not some magic string